### ISSUE TITLE
Возможность загружать глобальные переменные из внешнего файла

### ISSUE THREAD

**Issue by ripreal** [Pr-Mex/vanessa-automation#60]
> Заполните шаблон обсуждения GIT

```Gherkin
Функционал: Загрузка глобальных переменных из файла
   Как разработчиков тестов 
   Я хочу иметь конфигурационый файл, в котором я запишу все плавающие пользовательские переменные (имена баз, строки подключений, логины, пароли) , чтобы не влезать в фичи каждый раз, если я хочу их поменять

Контекст:
  Дано: Vanessa 
  И  8.3.12.1685

Сценарий:  Я перехожу в Ванессе на закладку Сервис и указываю файл user_settings.json из которого подтягивается контекст глобальных переменных. В дальнейшем в любой части теста я могу обращаться к этим переменным с помощью шагов, которые поддерживают работу с ними, например И в поле 'ИмяПоля' я ввожу значение глобальной переменной 'ИмяПеременной'
```

Примерный вариант файла user_settings.json
![image](https://user-images.githubusercontent.com/25801912/46474241-07637280-c7eb-11e8-81d6-3c1edfe296e5.png)

**Comment by ripreal** [comment]
Нам очень нужна эта функциональность, т.к. с фичами работает несколько людей. Эту функциональность мы реализовали в своих дописках в Ванессе. Хотелось бы их сконтрибьютить.

**Comment by Pr-Mex** [comment]
@ripreal 
Привет!
Интересная доработка!
А в реальном файле хранятся Настройки для всех пользователей? Или для каждого юзера свой файл?

**Comment by ripreal** [comment]
@Pr-Mex Да, это один файл, в котором содержатся настройки все пользователей. Этот файл хранится в репозитории проекта. Есть какие-нибудь особые правила по участию в проекте, могу ли я создать пулл реквест на эту задачу?

**Comment by Pr-Mex** [comment]
@ripreal 
>Есть какие-нибудь особые правила по участию в проекте, могу ли я создать пулл реквест на эту задачу?

Особых правил нет.  Смело создавай пулреквест.

Насчет данных из json, которые должны загрузиться в структуру глобальных переменных.
Я правильно понимаю, что в твоём примере это будет одна глобальная переменная "**userSettings**", внутри которой уже будет массив?

**Comment by ripreal** [comment]
@Pr-Mex Нет, userSettings - всего лишь того, что соответствовать правилам JSON API. Сами переменные у нас начинаются с ГЛ_ и сохраняются непосредственно в "ОбъектКонтекстСохраняемый". 

Порядок работы с ними следующий:

1. Пользователь у себя в каталоге с обработкой ванессы создает файл user_settings.json. Сам файл user_settings.json должен отвечать специальному формату: он должен называться именно так и содержать структуру как в примере выше. Но сами имена переменных могут быть любыми.
2. Когда открывается обработка Vanessa Behavior, файл user_settings.json подгятивается автоматически, берется он из каталога проекта (поле Каталог проекта на вкладке Сервис) Если такого файла нет, то загрузка молча игнорируется.
3. Если файл найден, то на основании текущего имени пользователя компьютера или домена (которое определяется через WShell скрипт), ищутся настройки текущего пользователя и загружаются только они.

Можно конечно доработать механизм и вынести на вкладку "Сервис" ванессы отдельное поле с указанием пути к загружаемым настройкам. Но у нас в проекте - это лишняя функциональность. Мы бы ей не пользовались.

**Comment by Pr-Mex** [comment]
Тогда, возможно, логичнее сделать "точку расширения" в Ванессе.
Например, в процедуре ПриОткрытии() передавать управление твоей обработке, которая будет вычитывать эти файлы по тем правилам как вам удобно.
Сейчас эта функциональность получается заточена под ваш флоу.

С точкой расширения вы сможете кастомизировать запуск Ванессы, и спокойно обновлять новые версии.

**Comment by Pr-Mex** [comment]
@ripreal 
Я имел ввиду точки расширения по аналогии с EDT.

>Почему бы не добавить эту функциональность в саму ванессу? Мне кажется она могла быть полезна многим. И я готов ее доработать, чтобы сделать ее более универсальной, если это нужно.

Ок. Давай.

**Comment by ripreal** [comment]
@Pr-Mex Заметил, что допустил мелкую ошибку в readme.md. В блоке где приведен образец формата user-settings.json нужно заменить символ окончания объекта } на символ окончания массива ]
